lib/Map/UserDrawing.js

```diff
@@ -0,0 +1,400 @@
+'use strict';
+
+import MapInteractionMode from '../Models/MapInteractionMode';
+
+var DeveloperError = require('terriajs-cesium/Source/Core/DeveloperError');
+var defined = require('terriajs-cesium/Source/Core/defined');
+var defaultValue = require('terriajs-cesium/Source/Core/defaultValue');
+var Color = require('terriajs-cesium/Source/Core/Color');
+var PolylineGlowMaterialProperty = require('terriajs-cesium/Source/DataSources/PolylineGlowMaterialProperty');
+var CustomDataSource = require('terriajs-cesium/Source/DataSources/CustomDataSource');
+var CallbackProperty = require('terriajs-cesium/Source/DataSources/CallbackProperty');
+var PolygonHierarchy = require('terriajs-cesium/Source/Core/PolygonHierarchy');
+var knockout = require('terriajs-cesium/Source/ThirdParty/knockout');
+var Entity = require('terriajs-cesium/Source/DataSources/Entity.js');
+var when = require('terriajs-cesium/Source/ThirdParty/when');
+var Cartesian3 = require('terriajs-cesium/Source/Core/Cartesian3');
+
+
+/**
+ * @alias UserDrawing
+ * @constructor
+ *
+ * For user drawings, which includes lines and/or a polygon
+ * @param {Terria} terria The Terria instance.
+ * @param {Object} options Object with the following properties, all of which are optional:
+ * @param {String} [options.messageHeader] Heading for the dialog which pops up when in user drawing mode.
+ * @param {Bool}   [options.allowPolygon] Let the user click on first point to close loop.
+ * @param {Object} [options.onPointClicked] Callback for when a point is clicked. Function that takes a CustomDataSource.
```

Comment:
JSDoc has syntax for documenting the signature of callbacks:
http://usejsdoc.org/tags-callback.html